src/document/DocumentCommandHandlers.js

```diff
@@ -1596,6 +1600,34 @@ define(function (require, exports, module) {
         ProjectManager.showInTree(MainViewManager.getCurrentlyViewedFile(MainViewManager.ACTIVE_PANE));
     }
 
+    /** Download selected file or folder structure **/
+    function handleFileDownload() {
+        var selectedItem = ProjectManager.getSelectedItem();
+        var path = selectedItem._path;
+        try {
+            fs.stat(path, function(err, stats) {
+                if (stats.type === "DIRECTORY") {
```

Comment:
You need to check `if(err) {...}` here, and probably log something or show a modal dialog indicating that there was a problem.